### PR TITLE
401 notification

### LINKED ISSUE #98: Inform user on 401 not authorized failed actions in UI
**Describe the bug**
When a user with the role "user" tries to do an action he is not authorized to do, it fails silently with no output (we get only 401 errors on the console).

**Affected Component(s)**
- [X] Back-End (Node.js, Next.js)
- [X] Front-End (React.js)
- [ ] Theming (CSS, React theme)
- [ ] Build System 
- [ ] Docs

**To Reproduce**
1. Try to to an unauthorized action on a hijack as a non-admin, or a multi-action on several hijacks.

**Expected behavior**
The user should be properly informed, preferable with a 401 not authorized error explicitly in the UI.
You can use either a popup or a page redirection.

**Screenshots**

**System (please complete the following information):**

**Additional context**
Security-wise it is fine, but the user should be informed in case sth is clickable but silently fails.

### PR DESCRIPTION
**Pull request by CuriouzK0d3r**
### Description of PR
Resolves #98 

Added notification in the hijack page + disabled the display of the actions the user does not has access to. 

What component(s) does this PR affect?

  - [ ] Back-End (Node.js, Next.js)
  - [x] Front-End (React.js)
  - [ ] Theming (CSS, React theme)
  - [ ] Build System
  - [ ] Docs
  
  
  
  Does the PR require changes on other components? If yes, please mark the components:
  
  - [ ] Back-End (Node.js, Next.js)
  - [ ] Front-End (React.js)
  - [ ] Theming (CSS, React theme)
  - [ ] Build System
  - [ ] Docs
  
  
  
### Related Issue
  
  
  
  
  
  
  Resolves #
  
### Type
  
  - [ ] Bug fix (non-breaking change which fixes an issue)
  - [ ] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to change)
  - [ ] Docs update
  - [ ] None of the above
  
  
  
### Checklist:
  
  - [ ] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
  - [ ] This change requires a change in the documentation.
  - [ ] I have updated the documentation accordingly.
  >)`>>>)>

### CODE REVIEW

File: apps/artemis-web/utils/fetch-data.ts [hunk 1]
```diff
@@ -144,6 +145,10 @@ export const sendHijackData = async (
     body: JSON.stringify(reqData),
   });
 
+  if (res.status === 403) {
+    toast("You do not have permission for this action!");
```
Comment (vkotronis) [inline apps/artemis-web/utils/fetch-data.ts:149]: what does `toast` output exactly? is it a popup, sth on console or sth else?

File: apps/artemis-web/utils/fetch-data.ts [hunk 2]
```diff
@@ -144,6 +145,10 @@ export const sendHijackData = async (
     body: JSON.stringify(reqData),
   });
 
+  if (res.status === 403) {
```
Comment (vkotronis) [inline apps/artemis-web/utils/fetch-data.ts:148]: please add also a `req.status == 401` condition too

File: apps/artemis-web/components/hijack-info/hijack-info.tsx [hunk 3]
```diff
@@ -182,59 +184,77 @@ class HijackInfoComponent extends Component<any, any> {
...
-                          </option>
-                        )}
-                        {!isSeen(this.props.hijackDataState) ? (
-                          <option value="hijack_action_acknowledge">
-                            Mark as Acknowledged
-                          </option>
+                            )}
+                            {!isSeen(this.props.hijackDataState) ? (
+                              <option value="hijack_action_acknowledge">
+                                Mark as Acknowledged
```
Comment (vkotronis) [inline apps/artemis-web/components/hijack-info/hijack-info.tsx:221]: The Ack and Not Ack actions should also be allowed to a non-admin user (it is OK with our ARTEMIS audience).
So a non-admin use should be able to ack/not ack, plus export hijack info. All the other actions are done by an admin, as you have implemented the check now.

**Review by vkotronis — CHANGES_REQUESTED** [review]
Please check and address the 3 comments in-PR. 
Moreover, please add role checks in the hijacks page: a non-admin user should only be able to ack/not ack hijacks in groups (talking about group actions on that page).

File: apps/artemis-web/utils/fetch-data.ts (hunk 1 above)
Comment (CuriouzK0d3r) [reply apps/artemis-web/utils/fetch-data.ts:149]: It is a notification (like alert box) on the top right of the screen :)

File: apps/artemis-web/pages/api/rmq_action.ts [hunk 4]
```diff
@@ -67,7 +67,7 @@ const sendRMQAction = async (obj) => {
 
 const handler = nc()
   .use(auth)
-  .use(authorization(['admin']))
+  .use(authorization(['admin', 'user']))
```
Comment (vkotronis) [inline apps/artemis-web/pages/api/rmq_action.ts:70]: what are the implications of this?

File: apps/artemis-web/components/hijack-info/hijack-info.tsx (hunk 3 above)
Comment (CuriouzK0d3r) [reply apps/artemis-web/components/hijack-info/hijack-info.tsx:221]: fixed :) To do that I changed the rmq_action endpoint permissions and added rmq_admin_action for the full set of actions

**Review by vkotronis — CHANGES_REQUESTED** [review]
Good job! Please add role checks in the hijacks page: a non-admin user should only be able to ack/not ack hijacks in groups (w.r.t. group actions on that page). Also, when I try to do sth with a simple user, like saving a comment or bulk-resolving hijacks, I get the 401 only in console and no toast is displayed. Could you check and verify please? After addressing these two comments, I can re-review and then we can merge. Thanks!

File: apps/artemis-web/components/hijack-table/hijack-table.tsx [hunk 5]
```diff
@@ -508,26 +509,46 @@ const HijackTableComponent = (props) => {
...
+            style={{
+              width: '200px',
+              display: 'inline-block',
+              marginRight: '5px',
+            }}
+            className="form-control form-control-sm-auto"
+            id="action_selection"
+          >
+            <option value="hijack_action_acknowledge">
+              Mark as Acknowledged
```
Comment (vkotronis) [inline apps/artemis-web/components/hijack-table/hijack-table.tsx:545]: the simple user should also be able to ack/not ack groups of hijacks. Currently (tested) he can non-ack, but not ack groups. Please check locally on test installation.

File: apps/artemis-web/pages/api/hijack.ts [hunk 6]
```diff
@@ -9,15 +9,26 @@ import { csrf } from '../../libs/csrf';
...
   .post(async (req: NextApiRequestExtended, res: NextApiResponseExtended) => {
     const host: string = process.env.DATABASE_HOST;
     const configHost: string = process.env.CONFIG_HOST;
     const port: number = parseInt(process.env.API_PORT, 10);
     let resp = null;
+    const allowed_normal_actions = ['hijack_action_acknowledge', 'hijack_action_acknowledge_not'];
 
     switch (req.body.action) {
       case 'comment':
+        if (req.user.role !== 'admin') {
```
Comment (vkotronis) [inline apps/artemis-web/pages/api/hijack.ts:22]: I tried to save a comment with simple user but nothing happens. Please check that there is some feedback on this. Same for the rest of the actions. I can demonstrate during our call too.

File: apps/artemis-web/components/hijack-table/hijack-table.tsx (hunk 5 above)
Comment (CuriouzK0d3r) [reply apps/artemis-web/components/hijack-table/hijack-table.tsx:549]: I think I fixed it now :) check again!

File: apps/artemis-web/pages/api/hijack.ts (hunk 6 above)
Comment (vkotronis) [reply apps/artemis-web/pages/api/hijack.ts:22]: This works (tested). Resolving.

File: apps/artemis-web/components/hijack-table/hijack-table.tsx (hunk 5 above)
Comment (vkotronis) [reply apps/artemis-web/components/hijack-table/hijack-table.tsx:549]: yes it works. Resolving.

**Review by vkotronis — APPROVED** [review]
LGTM! Good job.

**Pull request merged**